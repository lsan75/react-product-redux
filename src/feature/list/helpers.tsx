import { IProduct } from '../../store/product/product'

export const productList: IProduct[] = [
  {
    name: 'aaa',
    price: 10,
    selected: true
  },
  {
    name: 'bbb',
    price: 20,
    selected: false
  },
  {
    name: 'ccc',
    price: 50
  }
]