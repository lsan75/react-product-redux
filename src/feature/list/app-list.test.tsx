import * as React from 'react'
import * as Redux from 'redux'
import * as createMockStore from 'redux-mock-store'

import * as Adapter from 'enzyme-adapter-react-16'

import { configure, mount, ReactWrapper, shallow, ShallowRendererProps, ShallowWrapper } from 'enzyme'

import { shallowToJson } from 'enzyme-to-json'

import { Provider } from 'react-redux'
import thunk from 'redux-thunk'

import ConnectedContainer, { IProps } from './app-list'

import { productList } from './helpers'

describe('AppList', () => {
  configure({ adapter: new Adapter() })

  const middleware: Redux.Middleware[] = [ thunk ]
  const mockStore = createMockStore(middleware)
  const store = mockStore({
    productReducer: {
      productList,
      totalPrice: 0
    }
  })
  const props: ShallowRendererProps = {
    context: { store }
  }
  let reactOutput: ReactWrapper<IProps>
  let shallowOutput: ShallowWrapper<IProps>


  beforeEach(() => {
    reactOutput = mount<IProps>(
      <Provider store={store}>
        <ConnectedContainer />
      </Provider>
    )
    shallowOutput = shallow<IProps>(
      <Provider store={store}>
        <ConnectedContainer />
      </Provider>
    ).dive(props)
  })

  afterEach(() => {
    store.clearActions()
  })

  it('renders without crashing', () => {
    const innerProps: IProps = shallowOutput.props()

    expect(shallowToJson(shallowOutput)).toMatchSnapshot()
    expect(innerProps.productList).toEqual(productList)
  })

  it('should select a product', () => {

    reactOutput.find('ul').childAt(0).simulate('click')

    const actions = store.getActions()
    expect(actions).toContainEqual({
      product: {
        name: 'aaa',
        price: 10,
        selected: true
      },
      type: 'PRODUCT_ACTION.SELECT_PRODUCT'
    })
  })
})