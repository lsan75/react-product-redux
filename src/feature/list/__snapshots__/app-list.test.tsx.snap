// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppList renders without crashing 1`] = `
<AppList
  getProducts={[Function]}
  productList={
    Array [
      Object {
        "name": "aaa",
        "price": 10,
        "selected": true,
      },
      Object {
        "name": "bbb",
        "price": 20,
        "selected": false,
      },
      Object {
        "name": "ccc",
        "price": 50,
      },
    ]
  }
  selectProduct={[Function]}
/>
`;
